pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/MLTransactionMetadataStore.java

```diff
@@ -400,6 +400,11 @@ public CompletableFuture<Void> updateTxnStatus(TxnID txnID, TxnStatus newStatus,
                             appendLogCount.increment();
                             try {
                                 synchronized (txnMetaListPair.getLeft()) {
+                                    if (txnMetaListPair.getLeft().status() == newStatus) {
+                                        transactionLog.deletePosition(Collections.singletonList(position));
+                                        promise.complete(null);
```

Comment:
if return, may we should endTransactionInTransactionBuffer twice and update committing to committed twice or once notFound right?

Reply:
Depending on how we design method `org.apache.pulsar.transaction.coordinator.impl.MLTransactionMetadataStore#updateTxnStatus`:
- If `updateTxnStatus` is idempotent, then we should use this PR to add concurrent control to return success when `txnMetaListPair.getLeft().status() == newStatus`. 
- If `updateTxnStatus`  is not  idempotent, then we should change line 386 from
```
promise.complete(null);
```
to
```
promise.completeExceptionally(e);
```
If `updateTxnStatus` is idempotent, we may issue commit/abort command to TB/TP multiple times, as a result TB/TP have to handle the commit/abort command idempotently.